pd-client/client.go

```diff
@@ -430,27 +434,38 @@ var tsoReqPool = sync.Pool{
 	},
 }
 
-func (c *client) GetTS(ctx context.Context) (int64, int64, error) {
-	start := time.Now()
-	defer func() { cmdDuration.WithLabelValues("tso").Observe(time.Since(start).Seconds()) }()
-
+func (c *client) GetTSAsync(ctx context.Context) *tsoRequest {
 	req := tsoReqPool.Get().(*tsoRequest)
+	req.start = time.Now()
+	req.ctx = ctx
```

Comment:
It should be here, you can't just consider the cancel case.
Imagine that you need pass some value to the callee, it's too late to put it in Wait.